docs/bucket/notification/README.md

```diff
@@ -0,0 +1,113 @@
+
+- _ObjectRemoved_ event is generated whenever an object is deleted.
+
+| Event types| Description|
+|:-----------|:-----------|
+|_s3:ObjectRemoved:*_| All delete operations.|
+|_s3:ObjectRemoved:Delete_||
+
+## Supported Targets
+
+Minio can send bucket notifications to one or more of the following supported targets. The target is identified uniquely through an unique resource identifier called [ARN](http://docs.aws.amazon.com/general/latest/gr/aws-arns-and-namespaces.html).
+
+### AMQP via Simple Queue Service (SQS)
+
+The Advanced Message Queuing Protocol (AMQP) is an open standard application layer protocol for message-oriented middleware. The defining features of AMQP are message orientation, queuing, routing (including point-to-point and publish-and-subscribe), reliability and security.
+
+Configure `~/.minio/config.json` with your AMQP config.
+
+```json
+	"notify": {
+		"amqp": {
+			"1": {
+				"enable": true,
+				"url": "amqp://minio:minio@localhost:5672/",
+				"exchange": "minio1_direct",
+				"routingKey": "minio1_direct",
+				"exchangeType": "direct",
+				"mandatory": false,
+				"immediate": false,
+				"durable": true,
+				"internal": false,
+				"noWait": false,
+				"autoDeleted": false
+			}
+		},
+        }
+```
+
+### Elasticsearch via Simple Queueing Service (SQS)
+
+Elasticsearch is a search engine based on Lucene. It provides a distributed, multitenant-capable full-text search engine with an HTTP web interface and schema-free JSON documents. To know more details about configuration please read [Minio server configuration guide](https://docs.minio.io/docs/minio-server-configuration-files-guide)
+
+Configure `~/.minio/config.json` with your ElasticSearch config.
+
+```json
+	"notify": {
+		"elasticsearch": {
+			"1": {
+				"enable": true,
+				"url": "http://localhost:9200",
+                                "index": "minio"
+			}
+		},
+        }
+```
+
+### Redis via Simple Queueing Service (SQS)
+
+Redis is a data structure server. It is open-source, networked, in-memory, and stores keys with optional durability. To know more details about configuration please read [Minio server configuration guide](https://docs.minio.io/docs/minio-server-configuration-files-guide)
+
+Configure `~/.minio/config.json` with your Redis config.
+
+```json
+	"notify": {
+		"redis": {
+			"1": {
+				"enable": true,
+				"address": "localhost:6379",
+                                "password": "minio",
+                                "key": "minio1"
+			}
+		},
+        }
+```
+
+### Listen via Simple Notification Service (SNS)
+
+Applications may directly listen for events using an extended [ListenBucketNotification RESTful API](https://docs.minio.io/docs/golang-client-api-reference#ListenBucketNotification).
+
+You may also look at [mc watch](https://docs.minio.io/docs/minio-client-complete-guide#watch) command to write useful shell scripts around bucket events.
```

Comment:
We are calling event-notifications by different names within this document. We need a uniform convention for this.